src/ecalc/libraries/libecalc/common/libecalc/core/graph_result.py

```diff
@@ -230,30 +236,89 @@ def get_asset_result(self) -> libecalc.dto.result.EcalcModelResult:
                 ]
             )
 
-            sub_components.append(
-                parse_obj_as(
-                    libecalc.dto.result.ComponentResult,
-                    {
-                        **consumer_result.component_result.dict(),
-                        "name": consumer_node_info.name,
-                        "parent": self.graph.get_predecessor(consumer_id),
-                        "component_level": consumer_node_info.component_level,
-                        "componentType": consumer_node_info.component_type,
-                        "emissions": self._parse_emissions(self.emission_results[consumer_id])
-                        if consumer_id in self.emission_results
-                        else [],
-                        "energy_usage_cumulative": consumer_result.component_result.energy_usage.to_volumes()
+            if self.graph.get_component(consumer_id).component_type == ComponentType.COMPRESSOR:
+                component = self.graph.get_component(consumer_id)
+
+                # Get temporal expression for inlet- and outlet pressures
+                inlet_pressure_eval = CompressorResult.evaluate_energy_usage_model(
+                    energy_usage_model=component.energy_usage_model,
+                    pressure_type=CompressorPressureState.INLET_PRESSURE,
+                )
+
+                outlet_pressure_eval = CompressorResult.evaluate_energy_usage_model(
+                    energy_usage_model=component.energy_usage_model,
+                    pressure_type=CompressorPressureState.OUTLET_PRESSURE,
+                )
+
+                # Evaluate pressures in temporal expression
+                requested_inlet_pressure = TimeSeriesFloat(
+                    timesteps=self.timesteps,
+                    values=TemporalExpression.evaluate(inlet_pressure_eval, self.variables_map),
+                    unit=Unit.BARA,
+                )
+
+                requested_outlet_pressure = TimeSeriesFloat(
+                    timesteps=self.timesteps,
+                    values=TemporalExpression.evaluate(outlet_pressure_eval, self.variables_map),
+                    unit=Unit.BARA,
+                )
+
+                sub_components.append(
+                    CompressorResult(
+                        componentType=consumer_node_info.component_type,
+                        component_level=consumer_node_info.component_level,
+                        name=consumer_node_info.name,
+                        timesteps=self.timesteps,
+                        isValid=consumer_result.component_result.is_valid,
+                        energyUsage=consumer_result.component_result.energy_usage,
+                        energyUsageCumulative=consumer_result.component_result.energy_usage.to_volumes()
                         .cumulative()
                         .dict(),
-                        "power_cumulative": consumer_result.component_result.power.to_volumes()
```

Comment:
removed by mistake?

Reply:
Not removed, it is here: https://github.com/equinor/ecalc/blob/a840166e089ae75bd384952e513a7dd52f4eb396/src/ecalc/libraries/libecalc/common/libecalc/core/graph_result.py#L281